### ISSUE TITLE
Python/Issue with Calculating Total USDT Transactions on Tron Blockchain via API

### ISSUE THREAD

**Issue by 808culture** [tronprotocol/tronweb#470]
I am developing a Python script intended to calculate the total amount of incoming USDT (TRC20) transactions to a specific address on the Tron blockchain. The script is supposed to utilize either the TronGrid or TronScan API to fetch transaction data. However, upon executing the script, it returns the total sum as 0, which contradicts the expected results, as there are incoming USDT transactions on the address.

I have already tried various approaches and APIs, but none have yielded the correct outcome. Here's the script I'm using 

```
import requests

def fetch_transactions(address, start=0, limit=50):
    url = f"https://api.trongrid.io/v1/accounts/{address}/transactions/trc20?limit={limit}&only_to=true&min_block_timestamp=0&only_confirmed=true"
    response = requests.get(url)
    if response.status_code == 200:
        return response.json().get("data", [])
    else:
        print(f"Failed to fetch transactions for address: {address}")
        return []

def get_total_usdt(address):
    transactions = fetch_transactions(address)
    total_usdt = 0

    for tx in transactions:
        if tx.get('token_info', {}).get('symbol') == 'USDT':
            total_usdt += int(tx.get('value', 0)) / 10**6  # Adjusting for token's decimal

    return total_usdt

# Example usage (replace 'your_address' with the actual address)
address = "your_address"
total_usdt = get_total_usdt(address)
print(f"Total incoming USDT transactions for address {address}: {total_usdt}")

```

I have encountered the following issues:

The script always returns a total of 0 USDT, despite there being incoming transactions.
I'm unsure if the problem lies with the way I'm using the API, the API itself, or the script.
Any advice or suggestions on how to resolve this issue would be greatly appreciated. Thank you!

**Comment by 808culture** [comment]
import requests

def fetch_transactions(address, start=0, limit=50):
    url = f"https://api.trongrid.io/v1/accounts/{address}/transactions/trc20?limit={limit}&only_to=true&min_block_timestamp=0&only_confirmed=true"
    response = requests.get(url)
    if response.status_code == 200:
        return response.json().get("data", [])
    else:
        print(f"Failed to fetch transactions for address: {address}")
        return []

def get_total_usdt(address):
    transactions = fetch_transactions(address)
    total_usdt = 0

    for tx in transactions:
        if tx.get('token_info', {}).get('symbol') == 'USDT':
            total_usdt += int(tx.get('value', 0)) / 10**6  # Adjusting for token's decimal

    return total_usdt

# Example usage (replace 'your_address' with the actual address)
address = "your_address"
total_usdt = get_total_usdt(address)
print(f"Total incoming USDT transactions for address {address}: {total_usdt}")

**Comment by 808culture** [comment]
import requests

def fetch_transactions(address, start=0, limit=50):
    url = f"https://api.trongrid.io/v1/accounts/{address}/transactions/trc20?limit={limit}&only_to=true&min_block_timestamp=0&only_confirmed=true"
    response = requests.get(url)
    if response.status_code == 200:
        return response.json().get("data", [])
    else:
        print(f"Failed to fetch transactions for address: {address}")
        return []

def get_total_usdt(address):
    transactions = fetch_transactions(address)
    total_usdt = 0

    for tx in transactions:
        if tx.get('token_info', {}).get('symbol') == 'USDT':
            total_usdt += int(tx.get('value', 0)) / 10**6  # Adjusting for token's decimal

    return total_usdt

# Example usage (replace 'your_address' with the actual address)
address = "your_address"
total_usdt = get_total_usdt(address)
print(f"Total incoming USDT transactions for address {address}: {total_usdt}")

**Comment by 808culture** [comment]
```
import requests

def fetch_transactions(address, start=0, limit=50):
    url = f"https://api.trongrid.io/v1/accounts/{address}/transactions/trc20?limit={limit}&only_to=true&min_block_timestamp=0&only_confirmed=true"
    response = requests.get(url)
    if response.status_code == 200:
        return response.json().get("data", [])
    else:
        print(f"Failed to fetch transactions for address: {address}")
        return []

def get_total_usdt(address):
    transactions = fetch_transactions(address)
    total_usdt = 0

    for tx in transactions:
        if tx.get('token_info', {}).get('symbol') == 'USDT':
            total_usdt += int(tx.get('value', 0)) / 10**6  # Adjusting for token's decimal

    return total_usdt

# Example usage (replace 'your_address' with the actual address)
address = "your_address"
total_usdt = get_total_usdt(address)
print(f"Total incoming USDT transactions for address {address}: {total_usdt}")

```

**Comment by alextrader007** [comment]
I used your script locally to test that everything is normal. You can directly request the Trongrid interface in the browser to see if there is data returned.<img width="1049" alt="image" src="https://github.com/tronprotocol/tronweb/assets/129820300/8a83860e-7695-47b5-a25e-034078c38730">

**Comment by alextrader007** [comment]
The only thing to note is that some address records may be larger than 50. You need to get all the other records. <img width="1059" alt="image" src="https://github.com/tronprotocol/tronweb/assets/129820300/97e2243d-43b8-42fe-94d2-c27f38ab6af8">
<img width="1084" alt="image" src="https://github.com/tronprotocol/tronweb/assets/129820300/7fec197b-44f1-483e-a1d2-bcaaecb92a67">

**Comment by hughdaniels-sketch** [comment]
I never knew that my wife Had a son out of wedlock without informing me about it for almost 12years of our relationship until I noticed she picks strange calls on her phone and send money to a strange man. As much as it is inappropriate to spy on your spouse  it's also necessary to be sure that your relationship is in perfect shape. All my gratitude goes to this Genuine Genius Hacker -: Cybertechie who got to clone/hack her phone completely to the extent that I had a clone of her phone and could read all her whatsapp & amp;Facebook message, call logs text message and GPS to know where and what she is always up to. I'm really grateful for this great experience, it really helped me find out the truth about my marriage. Cybertechie and his team also specialize on hacking and recovery of money/bitcoin lost to scammers online, kindly send a message if you're experiencing any of these issues to cybertechie813 @ gmail. com or WhatsApp +358 44- 9349-429.. Kindly reach out to them and thank me later.

**Comment by realbrianfletcher-a11y** [comment]
HAVE YOU LOST A FORTUNE TO CRYPTOCURRENCY INVESTMENT FRAUDSTERS, PIG BUTCHERING HOAX, PHISHING LINKS & ET CETERA ? HERE COMES REDEMPTION WITH EMERALD HACKS....SCHEDULE A CONSULTATION WITH AN EXPERT TODAY.

Emeraldhacks was founded by a small group of blockchain analysts and financial-crime investigators who spent years building tracing tools for exchanges and law enforcement. We saw the same pattern too often: a victim reports a scam, and the trail goes cold because no one follows it fast enough.

We built Emeraldhacks to close that gap. Every case is handled by an analyst, not a call-center script — and every case gets an honest read on whether recovery is realistic before you spend a dollar.

Explore our website below to get started-
emeraldhacks(.)netlify(.)app 

Be alert for fake hackers, stay vigilant.
